### PR TITLE
Update kernel version to 5.2.0-m4

### PR DESCRIPTION
**Pull request by daneshk**
This is to apply carbon kernel changes in 5.2.0-m4
Resolves: #261, #263

### CODE REVIEW

File: components/org.wso2.carbon.deployment.notifier/src/test/java/org/wso2/carbon/deployment/notifier/BaseTest.java [hunk 1]
```diff
@@ -31,6 +35,8 @@
      * Basedir for all file I/O.
      */
     public static String basedir;
+    private static final String OS_NAME_KEY = "os.name";
+    private static final String WINDOWS_PARAM = "indow";
```
Comment (Niranjan-K) [inline components/org.wso2.carbon.deployment.notifier/src/test/java/org/wso2/carbon/deployment/notifier/BaseTest.java]: this should be "windows"

File: features/org.wso2.carbon.deployment.engine.feature/resources/p2.inf [hunk 2]
```diff
@@ -0,0 +1,2 @@
+instructions.configure = \
+org.eclipse.equinox.p2.touchpoint.natives.mkdir(path:${installFolder}/../../deployment); \
```
Comment (Niranjan-K) [inline features/org.wso2.carbon.deployment.engine.feature/resources/p2.inf]: we would need to create two folders, i.e., one for the Server and then the other for the Runtime. Also can we check in windows, whether there is any issue if it tries to create the folder, if the folder already exists.

File: components/org.wso2.carbon.deployment.engine/pom.xml [hunk 3]
```diff
@@ -75,6 +75,16 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.utils</groupId>
+            <artifactId>carbon-utils</artifactId>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.config</groupId>
+            <artifactId>org.wso2.carbon.config</artifactId>
+            <optional>true</optional>
```
Comment (Niranjan-K) [inline components/org.wso2.carbon.deployment.engine/pom.xml]: If we have used import in the java class of these dependencies, then we need to remove the optional tag.

File: components/org.wso2.carbon.deployment.notifier/src/test/java/org/wso2/carbon/deployment/notifier/BaseTest.java (hunk 1 above)
Comment (imesh) [inline components/org.wso2.carbon.deployment.notifier/src/test/java/org/wso2/carbon/deployment/notifier/BaseTest.java]: The letter 'W' might need to be in uppercase or execute an ignore-case comparison.

Comment (daneshk) [inline components/org.wso2.carbon.deployment.notifier/src/test/java/org/wso2/carbon/deployment/notifier/BaseTest.java]: There can be case mismatch when we check for "windows", if os.name is "Windows ...". So I skipped the first letter.

Comment (daneshk) [inline components/org.wso2.carbon.deployment.notifier/src/test/java/org/wso2/carbon/deployment/notifier/BaseTest.java]: I will change this to "Windows".

File: features/org.wso2.carbon.deployment.engine.feature/resources/p2.inf (hunk 2 above)
Comment (daneshk) [inline features/org.wso2.carbon.deployment.engine.feature/resources/p2.inf]: Checked in windows. deployment directory creates without any errors.

File: components/org.wso2.carbon.deployment.engine/src/main/java/org/wso2/carbon/deployment/engine/config/DeploymentConfiguration.java [hunk 4]
```diff
@@ -40,10 +40,7 @@
     private DeploymentNotifierConfig deploymentNotifier = new DeploymentNotifierConfig();
 
     public DeploymentConfiguration() {
-        repositoryLocation = "${carbon.home}/deployment/";
-        if (Utils.getSystemVariableValue("carbon.home", null) != null) {
-            repositoryLocation = Utils.substituteVariables(repositoryLocation);
-        }
+        repositoryLocation = ConfigurationUtils.substituteVariables("${carbon.home}/deployment/");
```
Comment (Niranjan-K) [inline components/org.wso2.carbon.deployment.engine/src/main/java/org/wso2/carbon/deployment/engine/config/DeploymentConfiguration.java]: We need to monitor all the deployment folders, i.e., the Server deployment folder and the current runtime's deployment folder.

File: components/org.wso2.carbon.deployment.notifier/pom.xml [hunk 5]
```diff
@@ -36,6 +36,16 @@
...
     <dependencies>
         <dependency>
+            <groupId>org.wso2.carbon.utils</groupId>
+            <artifactId>carbon-utils</artifactId>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.config</groupId>
+            <artifactId>org.wso2.carbon.config</artifactId>
+            <optional>true</optional>
```
Comment (Niranjan-K) [inline components/org.wso2.carbon.deployment.notifier/pom.xml]: Do we need the optional tag?

Comment (Niranjan-K) [inline components/org.wso2.carbon.deployment.notifier/pom.xml]: Do we need the optional tag?

**Comment by Niranjan-K** [comment]
@daneshk : We would need to add test case to verify the deployment folders are working in both Server and Runtime deployment folders. As this is a change which is along with Carbon 5.2.0-m4

**Comment by msmshariq** [comment]
I have created issue #267 to track this, can we go ahead and merge this PR?

**Pull request merged**